MoviesExample/app/src/main/java/com/novoda/demo/movies/MoviesAdapter.java

```diff
@@ -97,64 +100,30 @@ public void bind(final Movie movie) {
     static class LoadPageItem extends RecyclerView.ViewHolder {
 
         MoviesListCardBinding binding;
-        Listener listener;
 
-        LoadPageItem(MoviesListCardBinding binding, Listener listener) {
+        LoadPageItem(MoviesListCardBinding binding) {
             super(binding.getRoot());
             this.binding = binding;
-            this.listener = listener;
         }
 
-        public void bind(final int page) {
+        public void bind() {
             Movie movie = new Movie();
-            movie.title = "Loading page " + page;
+            movie.title = "Loading next page";
```

Comment:
The information about the current page got lost here. In order to maintain this feature, we would need to expose it similar to the `NetworkStatus`.